jr-objects/src/main/java/com/fasterxml/jackson/jr/ob/impl/BeanPropertyIntrospector.java

```diff
@@ -182,4 +177,13 @@ private static String decap(String name) {
         return name;
     }
 
+    /**
+     * Another helper method to deal with Groovy's problematic metadata accessors
+     *
+     * @implNote Groovy MetaClass have cyclic reference, and hence the class containing it should not be serialised without
+     * either removing that reference, or skipping over such references.
+     */
+    protected static boolean isGroovyMetaClass(Class<?> clazz) {
+        return clazz.getName().startsWith("groovy.lang");
```

Comment:
This needs to check for class name, not just package, right?

Reply:
Initially i implemented it to check for just MetaClassImpl, but after checking jackson-databind, i just replicated its exclusion.

https://github.com/FasterXML/jackson-databind/blob/db95863d6f48cb5695bf8e14af1e7af2ece8e52e/src/main/java/com/fasterxml/jackson/databind/util/BeanUtil.java#L205

is this correct? or should we go for targeted exclusions?

Reply:
understood, I missed this code, will implement in a similar fashion.